Docs: add known issue 65488 - SQL ignoring non-eq

bpintea opened the pull request:
Add description of the #65488 as a known issue in the release notes.

bpintea committed:
Docs: add known issue 65488 - SQL ignoring non-eq

Add description of the #65488 known issue in release notes.

elasticmachine commented:
Pinging @elastic/es-docs (Team:Docs)

jrodewig commented on docs/reference/release-notes/7.10.asciidoc:
```diff
@@ -3,6 +3,16 @@
 
 Also see <<breaking-changes-7.10,Breaking changes in 7.10>>.
 
+[discrete]
+=== Known issues
```
Adds an explicit anchor to avoid auto-gen ones.

```suggestion
[[known-issues-7.10.0]]
[discrete]
=== Known issues
```

jrodewig commented on docs/reference/release-notes/7.10.asciidoc:
```diff
@@ -3,6 +3,16 @@
+[discrete]
+=== Known issues
+
+* SQL: in case the WHERE clause of an SQL query contains at least two
+comparisons in a conjunction (AND) of which one is an inequality and the other
+is a not-equals, the later can get ignored. The workaround is to substitute the
+non-equals with a `NOT IN` operator.
+The defect has been fixed in {es} 7.10.1, the best resolution is to upgrade to
+that or a later version. More details can be found in issue: {es-issue}65488[#65488].
```
The `inequality` and `not-equal` language confused me and seems to contradict the [SQL docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-operators.html#sql-operators-inequality).

I tried to clarify and add an example:
```suggestion
* SQL: If a `WHERE` clause contains a comparison (`<=`, `<`, `>=`, `>`)
condition and an inequality (`!=`, `<>`) condition joined by `AND`, the SQL
query ignores the inequality condition. For example, the following query ignores
the `foo != 20` condition:
+
[source,sql]
----
SELECT my-field
FROM my-index-000001
WHERE foo > 1 AND foo != 20
----
+
We have fixed this issue in {es} 7.10.1 and later versions. For more details,
see {es-issue}65488[#65488].
```

bpintea replied:
Thanks for the noting the misalignment in therms, @jrodewig. 
I've rephrased the paragraph to better specify the occurring conditions and add back the workaround - which I find relevant and inline with other known issue entries.
I've also left the example out, since we'd then need a corrected example then too and the entry might get large. The GH issue gives examples for both (and links to the resolution too, for whoever would like even more details).
Please have a look and feel free to suggest any further improvements.

jrodewig reviewed (commented):
Thanks for putting this together, @bpintea! I left a suggestion that clarifies the language and adds an example. I'll take another look once that's resolved.

bpintea committed:
Update docs/reference/release-notes/7.10.asciidoc

Co-authored-by: James Rodewig <40268737+jrodewig@users.noreply.github.com>

bpintea committed:
Add the known issues to all affected releases

Add the known issue under following releasesȘ
- 7.7.0 & .1
- 7.8.0 & .1
- 7.9.0 & .1 & .2 & .3
- 7.10.0

jrodewig commented on docs/reference/release-notes/7.10.asciidoc:
```diff
@@ -3,6 +3,19 @@
+[discrete]
+=== Known issues
+
+* SQL: If a WHERE clause contains at least two relational operators joined by
```
Small nit. I'd apply this across all the changes.
```suggestion
* SQL: If a `WHERE` clause contains at least two relational operators joined by
```

bpintea replied:
Applied, thanks!

jrodewig reviewed (approved):
LGTM. Thanks for iterating on this @bpintea.

bpintea committed:
Apply review suggestions

s/WHERE/`WHERE`

bpintea committed:
Fix duplication of known issues-section

7.9.0 release had a "known issues" section already, place the SQL issue
under that one instead of opening a new one.

costin reviewed (approved):
LGTM

philkra reviewed (commented):
LGTM

bpintea commented:
@jrodewig, I was about to backport this change, but then not sure if necessary or beneficial[*] since _that_ "code" - i.e. the release notes in pre 7.10 branches - won't get used again anymore. Still, let me know please if the practice is to do so in these circumstances.

[*] ...as those branches will contain the note, but not the fix.

jrodewig commented:
@bpintea I think backporting only to 7.10 branches makes sense in that case. That approach also aligns with our [docs backporting practices](https://github.com/elastic/elasticsearch/tree/master/docs#backporting-doc-fixes).

jrodewig commented:
The docs team discussed our backport policy for known issues and decided to backport to any branch containing affected release notes. I [backported this change to 7.9, 7.8, and 7.7. I also opened https://github.com/elastic/elasticsearch/pull/73489 to clarify our backport policy.

bpintea commented:
thanks, @jrodewig!
